pkg/devfile/header.go

```diff
@@ -11,6 +11,7 @@ type DevfileHeader struct {
 	// +kubebuilder:validation:Pattern=^([2-9])\.([0-9]+)\.([0-9]+)(\-[0-9a-z-]+(\.[0-9a-z-]+)*)?(\+[0-9A-Za-z-]+(\.[0-9A-Za-z-]+)*)?$
 	SchemaVersion string `json:"schemaVersion"`
 
+	// +kubebuilder:pruning:PreserveUnknownFields
```

Comment:
@yangcao77 Though this PR is already merged, I just have a question about the reason for adding this line. It would allow adding arbitrary fields directly inside the `metadata` object without any way to retrieve them from the API. Isn't it precisely the situation we fixed by adding `attributes` one level down ?

Reply:
Without this line,  the schema validation is going to fail when there are additional properties in metadata. 

The devfile registries for Odo 2.0 have devfiles ( [example](https://github.com/odo-devfiles/registry/blob/0b0e73401e462324a8db32b60c625f56ad639391/devfiles/nodejs/devfile.yaml#L5-L6) ) contain additional properties in metadata.  Without this line, using latest Odo (integrated with latest parser) with a created project would cause failure.  Since Odo 2.0 has already released with those registries,  devfile 2.0 changes need to be backward compatible.

Reply:
OK, make perfectly sense. Thanks for the explanation !